submission/submission-core/src/main/kotlin/ac/uk/ebi/biostd/submission/domain/request/SubmissionRequestReleaser.kt

```diff
@@ -111,6 +119,54 @@ class SubmissionRequestReleaser(
         return releasedFile
     }
 
+    private suspend fun suppressRequest(
+        accNo: String,
+        request: SubmissionRequest,
+    ) {
+        val sub = request.submission
+        logger.info { "$accNo ${sub.owner} Started suppressing submission files over ${sub.storageMode}" }
+        suppressSubmissionFiles(sub)
+        logger.info { "$accNo ${sub.owner} Finished suppressing submission files over ${sub.storageMode}" }
+    }
+
+    private suspend fun suppressSubmissionFiles(sub: ExtSubmission) {
+        suspend fun suppressFile(reqFile: SubmissionRequestFile) {
+            when (val file = reqFile.file) {
+                is NfsFile -> {
+                    val released = reqFile.copy(file = suppress(sub, reqFile.index, file), status = SUPPRESSED)
+                    rqtService.updateRqtFile(released)
+                }
+
+                is FireFile -> {
+                    if (file.published) {
+                        val suppressed = reqFile.copy(file = suppress(sub, reqFile.index, file), status = SUPPRESSED)
+                        rqtService.updateRqtFile(suppressed)
+                    } else {
+                        rqtService.updateRqtFile(reqFile.copy(status = SUPPRESSED))
+                    }
+                }
+            }
+        }
+
+        supervisorScope {
+            filesRequestService
+                .getSubmissionRequestFiles(sub.accNo, sub.version, REUSED)
```

Comment:
all other operations include `.withIndex()` to them include logs with the current index. I found that usefull personally so I will suggest to include it here.

Reply:
I check all the other operations and they use `concurrently`. Could you please share an example?